third_party/move/move-compiler/src/expansion/translate.rs

```diff
@@ -2927,12 +2935,28 @@ fn bind(context: &mut Context, sp!(loc, pb_): P::Bind) -> Option<E::LValue> {
                 Some(DeprecatedItem::Struct),
             )?;
             let tys_opt = optional_types(context, ptys_opt);
-            let vfields: Option<Vec<(Field, E::LValue)>> = pfields
-                .into_iter()
-                .map(|(f, pb)| Some((f, bind(context, pb)?)))
-                .collect();
-            let fields = fields(context, loc, "deconstruction binding", "binding", vfields?);
-            EL::Unpack(tn, tys_opt, fields)
+            let mut dotdot = None;
+            let mut vfields = vec![];
+            for (i, pfield) in pfields.iter().enumerate() {
+                match &pfield.value {
+                    P::BindFieldOrDotdot_::FieldBind(field, pbind) => {
+                        let lval = bind(context, pbind.clone())?;
+                        vfields.push((field.clone(), lval));
+                    },
+                    P::BindFieldOrDotdot_::Dotdot => {
+                        if i != pfields.len() - 1 {
+                            context.env.add_diag(diag!(
+                                Syntax::UnexpectedToken,
+                                (pfield.loc, "`..` must be at the end in a field list")
+                            ));
```

Comment:
Do we want to return some placeholder here to avoid additional errors?  Where would a missing field error be generated?

Reply:
It seems reasonable to point out all misplaced `..`'s. The missing field error is generated in the expression builder, where name resolution is done and struct definitions are available.

Reply:
I was just worried about
```
    struct A { a:u64, b:u64, c:u64, ... x:u64, y:u64, z:u64 }
    ...
        let A { .., z } = ...
```
producing a warning about every single field being missing, in addition to the `..` not being at the end.

But it looks like you don't do that, somehow.

Reply:
Let's add the case suggested by @brmataptos as a test, so that we don't regress on this behavior.

Reply:
done